Subject: [PATCH] Prevent lost mTaskToReturnTo value for moving home

When the task moves to back in stack. find the last task to set to go home.

Sometimes, empty TaskRecord (without any Activities) can be exist in task history even though the task is over homestack task.
So, the last task could not set mTaskToReturn value to home type bacause of empty TaskRecord.

Find the real task that has not empty activities and prevent lost mTaskToReturnTo value for moving home.

Test: It can be reproduced with Chrome easily.
      1. Launch app that is using 'moveTaskToBack' as finishing.
         check the recent activity list. (should be listed on it)
      2. reboot device
      3. launch app again from the launcher.
      4. press back back key to finish activity

      App will use moveTaskToBack to finish itself, but Activity still remain.
      mTaskToReturn to value can not be changed even though it was started from launcher.

diff --git a/services/core/java/com/android/server/am/ActivityStack.java b/services/core/java/com/android/server/am/ActivityStack.java
index 3aaddd8..1c3c989 100644
--- a/services/core/java/com/android/server/am/ActivityStack.java
+++ b/services/core/java/com/android/server/am/ActivityStack.java
@@ -4512,7 +4512,7 @@
         int numTasks = mTaskHistory.size();
         for (int taskNdx = numTasks - 1; taskNdx >= 1; --taskNdx) {
             final TaskRecord task = mTaskHistory.get(taskNdx);
-            if (task.isOverHomeStack()) {
+            if (task.isOverHomeStack() && task.mActivities != null && task.mActivities.size() > 0) {
                 break;
             }
             if (taskNdx == 1) {
